### ISSUE TITLE
pygit2 ignoring passphrases in /etc/salt/master

### ISSUE THREAD

**Issue by jnials** [saltstack/salt#33075]
### Description of Issue/Question

pygit seems to be ignoring the passphrase when trying to communicate to a git repository over ssh.

I have tried this using both the global `gitfs_passphrase` and the target based `passphrase` options.

Additionally, if I left the passphrase set, but removed it from the ssh key, then it worked.  This indicates that the passphrase isn't being used at all, because that should have error'd.
### Setup

```
sudo apt-get install cmake python-dev python-pip \
libffi-dev libssh2-1-dev zlib1g-dev libssl-dev \
pkg-config libhttp-parser-dev -y

wget https://github.com/libgit2/libgit2/archive/v0.24.1.tar.gz
tar xzf v0.24.1.tar.gz
cd libgit2-0.24.1/
cmake .
make
sudo make install
sudo ldconfig
sudo pip install pygit2
```

[/etc/salt/master](https://github.com/saltstack/salt/files/250866/master.txt)
### Steps to Reproduce Issue

Salt master is set up on a vagrant instance.  No proxies otherwise.

```
salt-run -ldebug fileserver.update
```

[Salt run debug output](https://github.com/saltstack/salt/files/250868/salt-run.txt)
### Versions Report

No minions involved.

[`salt --versions-report`](https://github.com/saltstack/salt/files/250889/salt-version.txt)

**Comment by jfindlay** [comment]
@jnials, thanks for reporting.

**Comment by rallytime** [comment]
FYI @terminalmage

**Comment by terminalmage** [comment]
I can't reproduce this, I get the following error message in the log (NOTE: I've modified the ``log_fmt_logfile`` to include additional information):

```
18:48:56,228 [salt.utils.gitfs][1312][ERROR   ][8849] Error occured fetching gitfs remote 'git@github.com:terminalmage/gitfs-test1': Failed to authenticate SSH session: Callback returned error
```

**Comment by terminalmage** [comment]
I tested using libgit2 0.24.1 and pygit2 0.24.0.

**Comment by terminalmage** [comment]
I tested by commenting out the passphrase parameter, which resulted in the error message I posted. When I uncomment the passphrase parameter, authentication works as expected. This was tested both with the global and per-remote versions of these parameters.

Including a passphrase parameter when the key is not protected with a passphrase does not generate an error, but this is not Salt's fault. Salt is passing the provided authentication information to pygit2, and pygit2 is not generating an error. If you feel pygit2 should generate an error in these cases, then feel free to [file an report upstream](https://github.com/libgit2/pygit2/issues/new).

Also, the log information you posted doesn't indicate that there is any problem, there is no error anywhere.
